### DISCUSSION TITLE
Listener on multiple capsules updated from transaction

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#101] [Q&A] [answered]
Hi, I'm listening two capsules that are being updated together in a transaction. The listener it's getting executed only once. Is this normal? For me is good this behaviour, but I want to understand it and know if it is correct.

```dart
// Listening to data module
_cancelDataModuleListener = cc.listen((use) {
  use(votePersonalCachedCapsule);
  use(voteOverallCachedCapsule);

  forceUpdate();
});
```

I ask this because I were expecting it to be executed twice.

**Comment by GregoryConrad** [comment, marked as answer]
> Is this normal? For me is good this behaviour, but I want to understand it and know if it is correct.

Yes! That is entirely the point of transactions--to update multiple possibly unrelated points of data together into one rebuild sweep. The singular rebuild resulting from a transaction is a behavior you can depend on and won't change.
